### PR TITLE
Reconciler Injection, now with code-gen!

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: codegen/cmd/injection-gen/generators/informer.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+// Key is used for associating the Informer inside the context.Context.
+type Key struct{}
+
+func withInformer(ctx context.Context) (context.Context, {{.controllerInformer|raw}}) {
+	f := {{.factoryGet|raw}}(ctx)
+	inf := f.{{.group}}().{{.version}}().{{.type|publicPlural}}()
+	return context.WithValue(ctx, Key{}, inf), inf.Informer()
+}
```
Comment (mattmoor-sockpuppet) [inline codegen/cmd/injection-gen/generators/informer.go]:
```suggestion
}
```

File: codegen/cmd/injection-gen/generators/informer.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+// Key is used for associating the Informer inside the context.Context.
+type Key struct{}
+
+func withInformer(ctx context.Context) (context.Context, {{.controllerInformer|raw}}) {
+	f := {{.factoryGet|raw}}(ctx)
+	inf := f.{{.group}}().{{.version}}().{{.type|publicPlural}}()
+	return context.WithValue(ctx, Key{}, inf), inf.Informer()
+}	
+
```
Comment (mattmoor-sockpuppet) [inline codegen/cmd/injection-gen/generators/informer.go]:
```suggestion

```

File: codegen/cmd/injection-gen/generators/informer.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (mattmoor-sockpuppet) [inline codegen/cmd/injection-gen/generators/informer.go]:
```suggestion
Copyright 2019 The Kubernetes Authors.
```

**Comment by mattmoor** [comment]
This is rebased, but I have a handful more things I'd like to add.
 - section to README.md on codegen usage
 - lint fixes to the codegen
 - logging to the codegen

**Pull request merged**